src/List.js

```diff
@@ -66,13 +86,33 @@ export function List() {
 
   const handleChange = async (id, event) => {
     let date = new Date();
+    const item = items.find((element) => element.id === id);
+    const daysSinceLastTransaction = item.lastPurchasedDate
+      ? Math.round((new Date() - item.lastPurchasedDate) / 1000 / 60 / 60 / 24)
+      : 0;
     const checked = event.target.checked;
     if (checked) {
       const itemRef = doc(db, 'shopping-list', id);
-      setDoc(itemRef, { lastPurchasedDate: date.getTime() }, { merge: true });
+      setDoc(
+        itemRef,
+        {
+          lastPurchasedDate: date.getTime(),
+          estimatedPurchaseDate: calculateEstimate(
+            item.estimatedPurchaseDate,
+            daysSinceLastTransaction,
+            item.totalPurchases,
+          ),
+          totalPurchases: item.totalPurchases + 1,
+        },
+        { merge: true },
+      );
     } else {
       const itemRef = doc(db, 'shopping-list', id);
-      setDoc(itemRef, { lastPurchasedDate: null }, { merge: true });
+      setDoc(
+        itemRef,
+        { lastPurchasedDate: null, totalPurchases: item.totalPurchases - 1 },
```

Comment:
if the user click unchecked and `lastPurchasedDate` changes to  `null`  we'll lose the information from  the `lastPurchasedDate` before it was unchecked?

Reply:
One thing to consider is whether a user should be allowed to uncheck? In the past groups have made the decision to not allow users to uncheck the checkbox.

Reply:
Thank you, we removed the ability to uncheck items so that we're in keeping with all the other groups!